tests/mock_crust/mod.rs

```diff
@@ -54,6 +75,51 @@ fn disconnect_on_rebootstrap() {
     expect_next_event!(unwrap!(nodes.last_mut()), Event::Terminated);
 }
 
+#[test]
+fn candidate_timeout_resource_proof() {
+    let network = Network::new(MIN_SECTION_SIZE, None);
+    let mut nodes = create_connected_nodes_until_split(&network, vec![1, 1], false);
```

Comment:
why do we need multiple sections here? sure I can reason with that in saying "it does check all nodes have lost a candidate cos only one section should have accepted the candidate and then purged it", but that should maybe be a seperate test to confirm a joining node is only accepted to one section. 

I'm just a bit partial about keeping smaller tests specific to what they're testing cos otherwise almost all tests would benefit from a bigger network setup ofc. not a biggie tho anyways if we wanted to leave as is.

Reply:
I was a bit confused initially by some of the logs and wondered whether the proxy node disconnection was because the node changed identity or whether we timed out the candidate.

As it is the test take 0.03 second on average to run on its own. So it seem reasonable to have a more representative flow.